BUG: nilearn.plotting forces matplotlib backend change to Agg on macOS

on macOS:
```
>>> import matplotlib
>>> assert matplotlib.get_backend() == 'Qt5Agg'
>>> import nilearn.plotting
>>> assert matplotlib.get_backend() == 'Qt5Agg', matplotlib.get_backend()
AssertionError: agg
```
I have `backend : Qt5Agg\ninteractive : True` in `~/.matplotlib/matplotlibrc`. I assume it's due to this line:

https://github.com/nilearn/nilearn/blob/master/nilearn/plotting/__init__.py#L37

It seems like unwanted behavior that the backend is changed.

--- a/nilearn/plotting/__init__.py
+++ b/nilearn/plotting/__init__.py
@@ -30,11 +30,11 @@ def _set_mpl_backend():
         if 'inline' in current_backend or 'nbagg' in current_backend:
             return
         # Set the backend to a non-interactive one for unices without X
-        if (os.name == 'posix' and 'DISPLAY' not in os.environ
-                and not (sys.platform == 'darwin'
-                         and 'macosx' in current_backend)
-        ):
+        # (see gh-2560)
+        if (sys.platform not in ('darwin', 'win32') and
+                'DISPLAY' not in os.environ):
             matplotlib.use('Agg')
+
 
 _set_mpl_backend()
 

